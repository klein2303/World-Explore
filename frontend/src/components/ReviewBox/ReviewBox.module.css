@media (min-width: 601px) {
    .reviewBox {
        position: relative;
        background-color: #fcf5ff;
        margin-top: 30px;
        margin-bottom: 30px;
        max-width: 100%;
        height: 290px;
        border: none;
        border-radius: 10px;
        resize: none;
        padding: 20px;
        padding-bottom: 60px;
    }

    .reviewTitle {
        font-size: 25px;
        width: fit-content;
    }

    .date {
        font-size: 15px;
        width: fit-content;
    }

    .reviewText {
        margin-top: 10px;
        position: absolute;
        background-color: #fcf5ff;
        width: 97%;
        height: 180px;
        font-size: 16px;
        overflow-y: scroll;
        white-space: pre-wrap;
        line-height: 2;
    }

    .starsContainer {
        position: absolute;
        left: 10px;
        bottom: 55px;
        display: flex;
        gap: 10px;
        width: fit-content;
        font-size: 35px;
        color: #f5d103;
    }

    .publicContainer {
        position: absolute;
        display: flex;
        flex-direction: row;
        bottom: 20px;
        gap: 10px;
        width: fit-content;
    }

    .checkBox {
        font-size: 25px;
    }

    .checkBox:hover {
        cursor: pointer;
    }

    .noJournal {
        font-size: 32px;
        font-style: italic;
        margin-top: 50px;
        text-align: center;
    }
}

@media (max-width: 500px) {
    /*Phone */
    .reviewBox {
        position: relative;
        background-color: #fcf5ff;
        margin-top: 30px;
        margin-bottom: 30px;
        max-width: 100%;
        height: 320px;
        border: none;
        border-radius: 10px;
        resize: none;
        padding: 20px;
        padding-bottom: 60px;
    }
    .reviewTitle {
        font-size: 25px;
        width: fit-content;
    }

    .date {
        font-size: 15px;
        width: fit-content;
    }

    .reviewText {
        margin-top: 10px;
        width: 100%;
        background-color: #fcf5ff;
        height: 180px;
        font-size: 16px;
        overflow-y: scroll;
        white-space: pre-wrap;
        line-height: 2;
    }

    .starsContainer {
        position: absolute;
        left: 10px;
        bottom: 55px;
        display: flex;
        gap: 10px;
        width: fit-content;
        font-size: 35px;
        color: #f5d103;
    }

    .publicContainer {
        position: absolute;
        display: flex;
        flex-direction: row;
        bottom: 20px;
        gap: 10px;
        width: fit-content;
    }

    .checkBox {
        font-size: 25px;
    }

    .checkBox:hover {
        cursor: pointer;
    }

    .noJournal {
        font-size: 24px;
        font-style: italic;
        margin-top: 70px;
        text-align: center;
    }
}
