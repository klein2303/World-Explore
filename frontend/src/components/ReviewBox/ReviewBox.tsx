import styles from "./ReviewBox.module.css";
import { FaRegStar, FaStar } from "react-icons/fa";
import { gql, useQuery } from "@apollo/client";
import { removeQuotes } from "../../utils/utils";

interface journalCountry {
    country: string;
}

const ReviewBox = ({ country }: journalCountry) => {
    const user = removeQuotes(sessionStorage.getItem("user")!);
    const JOURNAL = gql`
        query GetJournal($countryid: String!, $profileid: String!) {
            writtenjournal(countryid: $countryid, profileid: $profileid) {
                id
                reviews {
                    id
                    title
                    date
                    text
                    rating
                    ispublic
                }
            }
        }
    `;
    const { data, loading, error } = useQuery(JOURNAL, {
        variables: { countryid: country, profileid: user },
        fetchPolicy: "cache-first", // Used for first execution
        nextFetchPolicy: "cache-first", // Used for subsequent executions
    });
    // Render 5 stars based on a given rating
    const renderStars = (rating: number) => {
        const maxStars = 5;
        return Array.from({ length: maxStars }, (_, index) =>
            index < rating ? (
                <FaStar key={index} className={styles.starFilled} aria-label="filled star" />
            ) : (
                <FaRegStar key={index} className={styles.starNotFilled} aria-label="empty star" />
            ),
        );
    };

    if (loading) return <p>Loading...</p>;
    if (error) return <p>Error: {error.message}</p>;
    return (
        <main className="main">
            {data.writtenjournal ? (
                data.writtenjournal.reviews.map(
                    (review: {
                        id: number;
                        title: string;
                        date: string;
                        text: string;
                        rating: number;
                        ispublic: boolean;
                    }) => (
                        <section
                            key={review.id}
                            className={styles.reviewBox}
                            aria-description="Section containing the journal entry"
                            tabIndex={0}>
                            <p className={styles.reviewTitle}>{review.title}</p>
                            <p className={styles.date}>{review.date}</p>
                            <article className={styles.reviewText}>{review.text} </article>
                            <section className={styles.starsContainer} aria-label="rating">
                                {/* Render stars based on rating */}
                                {renderStars(review.rating)}
                            </section>
                            <p className={styles.publicInfoText}>
                                This journal entry is {review.ispublic ? "public" : "private"}
                            </p>
                        </section>
                    ),
                )
            ) : null}
        </main>
    );
};
export default ReviewBox;
