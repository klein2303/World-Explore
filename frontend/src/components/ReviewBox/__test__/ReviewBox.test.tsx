import { render, screen, waitFor } from "@testing-library/react";
import { describe, it, expect, vi, beforeEach } from "vitest";
import { MockedProvider } from "@apollo/client/testing";
import gql from "graphql-tag"; // Import gql separately if not from Apollo Client
import ReviewBox from "../ReviewBox";

vi.mock("../../utils/utils", () => ({
    removeQuotes: (value: string) => value,
}));

const JOURNAL = gql`
    query GetJournal($countryid: String!, $profileid: String!) {
        writtenjournal(countryid: $countryid, profileid: $profileid) {
            id
            reviews {
                id
                title
                date
                text
                rating
                ispublic
            }
        }
    }
`;

describe("ReviewBox", () => {
    const mockUser = "mockUserId";
    const consoleErrorSpy = vi.spyOn(console, "error").mockImplementation(() => {});

    beforeEach(() => {
        sessionStorage.setItem("user", mockUser);
    });

    afterEach(() => {
        consoleErrorSpy.mockRestore(); // Restore console.error after each test
    });

    const baseMock = {
        request: {
            query: JOURNAL,
            variables: { countryid: "France", profileid: mockUser },
        },
    };

    it("renders loading state initially", () => {
        const mocks = [
            {
                ...baseMock,
                result: {
                    data: null, // Simulate loading by not resolving immediately
                },
            },
        ];

        render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <ReviewBox country="France" />
            </MockedProvider>,
        );

        expect(screen.getByText(/loading.../i)).toBeInTheDocument();
    });

    it("renders error state when query fails", async () => {
        const mocks = [
            {
                request: {
                    query: JOURNAL,
                    variables: { countryid: "France", profileid: mockUser },
                },
                result: {
                    errors: [
                        {
                            message: "Query failed",
                            locations: [{ line: 2, column: 3 }],
                            path: ["writtenjournal"],
                        },
                    ],
                },
            },
        ];

        render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <ReviewBox country="France" />
            </MockedProvider>,
        );

        await waitFor(() => expect(screen.getByText(/error: query failed/i)).toBeInTheDocument());
    });

    it("renders reviews correctly when data is fetched", async () => {
        const mocks = [
            {
                ...baseMock,
                result: {
                    data: {
                        writtenjournal: {
                            id: "1",
                            reviews: [
                                {
                                    id: "101",
                                    title: "Amazing trip",
                                    date: "2024-11-20",
                                    text: "Loved the Eiffel Tower and the food!",
                                    rating: 5,
                                    ispublic: true,
                                },
                                {
                                    id: "102",
                                    title: "Relaxing vacation",
                                    date: "2024-11-15",
                                    text: "The countryside was peaceful and refreshing.",
                                    rating: 4,
                                    ispublic: false,
                                },
                            ],
                        },
                    },
                },
            },
        ];

        render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <ReviewBox country="France" />
            </MockedProvider>,
        );

        // Wait for the reviews to load
        await waitFor(() => expect(screen.getByText("Amazing trip")).toBeInTheDocument());

        // Verify the first review
        expect(screen.getByText("Amazing trip")).toBeInTheDocument();
        expect(screen.getByText("2024-11-20")).toBeInTheDocument();
        expect(screen.getByText("Loved the Eiffel Tower and the food!")).toBeInTheDocument();

        // Verify the second review
        expect(screen.getByText("Relaxing vacation")).toBeInTheDocument();
        expect(screen.getByText("2024-11-15")).toBeInTheDocument();
        expect(screen.getByText("The countryside was peaceful and refreshing.")).toBeInTheDocument();
    });

    it("renders no content if there are no reviews", async () => {
        const mocks = [
            {
                ...baseMock,
                result: {
                    data: {
                        writtenjournal: {
                            id: "1",
                            reviews: [],
                        },
                    },
                },
            },
        ];

        render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <ReviewBox country="France" />
            </MockedProvider>,
        );

        await waitFor(() => expect(screen.queryByText("Amazing trip")).not.toBeInTheDocument());
        expect(screen.queryByText(/relaxing vacation/i)).not.toBeInTheDocument();
    });

    it("renders stars correctly for different ratings", async () => {
        const mocks = [
            {
                ...baseMock,
                result: {
                    data: {
                        writtenjournal: {
                            id: "1",
                            reviews: [
                                {
                                    id: "103",
                                    title: "Mixed experience",
                                    date: "2024-10-10",
                                    text: "Some great moments, but also some issues.",
                                    rating: 3,
                                    ispublic: false,
                                },
                            ],
                        },
                    },
                },
            },
        ];

        render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <ReviewBox country="France" />
            </MockedProvider>,
        );

        // Wait for the data to load
        await waitFor(() => expect(screen.getByText("Mixed experience")).toBeInTheDocument());

        // Verify the star rendering for a 3-star review
        const ratingSection = screen.getByLabelText("rating");
        const stars = Array.from(ratingSection.childNodes);
        expect(stars).toHaveLength(5);

        const filledStars = stars.filter((star) => (star as Element).getAttribute("aria-label") === "filled star");
        expect(filledStars).toHaveLength(3); // 3 filled stars for the 3-star review
    });
});
