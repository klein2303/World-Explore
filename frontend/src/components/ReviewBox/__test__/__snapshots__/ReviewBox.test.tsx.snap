// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReviewBox Component > renders error state when query fails 1`] = `
<DocumentFragment>
  <p>
    Error: Query failed
  </p>
</DocumentFragment>
`;

exports[`ReviewBox Component > renders loading state initially 1`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;

exports[`ReviewBox Component > renders no reviews message when data contains no reviews 1`] = `
<DocumentFragment>
  <main
    class="main"
  >
    <section>
      <section
        class="_linkAndButton_756783"
      >
        <a
          class="_returnLink_756783"
          href="/MyJournals"
        >
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M200,32V176a8,8,0,0,1-8,8H67.31l34.35,34.34a8,8,0,0,1-11.32,11.32l-48-48a8,8,0,0,1,0-11.32l48-48a8,8,0,0,1,11.32,11.32L67.31,168H184V32a8,8,0,0,1,16,0Z"
            />
          </svg>
           Return to all journals
        </a>
        <button
          class="_countryPageButton_756783"
        >
          Go to info page
        </button>
      </section>
      <section
        class="_upperSection_756783"
      >
        <p
          aria-label="journals"
          class="_title_756783"
        >
          My France journals
        </p>
      </section>
    </section>
  </main>
</DocumentFragment>
`;

exports[`ReviewBox Component > renders reviews correctly when data is fetched 1`] = `
<DocumentFragment>
  <main
    class="main"
  >
    <section>
      <section
        class="_linkAndButton_756783"
      >
        <a
          class="_returnLink_756783"
          href="/MyJournals"
        >
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M200,32V176a8,8,0,0,1-8,8H67.31l34.35,34.34a8,8,0,0,1-11.32,11.32l-48-48a8,8,0,0,1,0-11.32l48-48a8,8,0,0,1,11.32,11.32L67.31,168H184V32a8,8,0,0,1,16,0Z"
            />
          </svg>
           Return to all journals
        </a>
        <button
          class="_countryPageButton_756783"
        >
          Go to info page
        </button>
      </section>
      <section
        class="_upperSection_756783"
      >
        <p
          aria-label="journals"
          class="_title_756783"
        >
          My France journals
        </p>
      </section>
    </section>
    <section
      aria-description="Section containing the journal entry"
      class="_reviewBox_756783"
      tabindex="0"
    >
      <section
        class="_iconcontainer_756783"
      >
        <svg
          aria-label="delete"
          class="_deleteIcon_756783"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          tabindex="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m112 112 20 320c.95 18.49 14.4 32 32 32h184c17.67 0 30.87-13.51 32-32l20-320"
            fill="none"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="32"
          />
          <path
            d="M80 112h352"
            stroke-linecap="round"
            stroke-miterlimit="10"
            stroke-width="32"
          />
          <path
            d="M192 112V72h0a23.93 23.93 0 0 1 24-24h80a23.93 23.93 0 0 1 24 24h0v40m-64 64v224m-72-224 8 224m136-224-8 224"
            fill="none"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="32"
          />
        </svg>
      </section>
      <p
        class="_reviewTitle_756783"
      >
        Amazing trip
      </p>
      <p
        class="_date_756783"
      >
        2024-11-20
      </p>
      <article
        class="_reviewText_756783"
      >
        Loved the Eiffel Tower and the food! 
      </article>
      <section
        aria-label="rating"
        class="_starsContainer_756783"
      >
        <svg
          aria-label="filled star"
          class="_starFilled_756783"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
        <svg
          aria-label="filled star"
          class="_starFilled_756783"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
        <svg
          aria-label="filled star"
          class="_starFilled_756783"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
        <svg
          aria-label="filled star"
          class="_starFilled_756783"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
        <svg
          aria-label="filled star"
          class="_starFilled_756783"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
      </section>
      <p
        class="_publicInfoText_756783"
      >
        This journal entry is public
      </p>
    </section>
    <section
      aria-description="Section containing the journal entry"
      class="_reviewBox_756783"
      tabindex="0"
    >
      <section
        class="_iconcontainer_756783"
      >
        <svg
          aria-label="delete"
          class="_deleteIcon_756783"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          tabindex="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m112 112 20 320c.95 18.49 14.4 32 32 32h184c17.67 0 30.87-13.51 32-32l20-320"
            fill="none"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="32"
          />
          <path
            d="M80 112h352"
            stroke-linecap="round"
            stroke-miterlimit="10"
            stroke-width="32"
          />
          <path
            d="M192 112V72h0a23.93 23.93 0 0 1 24-24h80a23.93 23.93 0 0 1 24 24h0v40m-64 64v224m-72-224 8 224m136-224-8 224"
            fill="none"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="32"
          />
        </svg>
      </section>
      <p
        class="_reviewTitle_756783"
      >
        Relaxing vacation
      </p>
      <p
        class="_date_756783"
      >
        2024-11-15
      </p>
      <article
        class="_reviewText_756783"
      >
        The countryside was peaceful and refreshing. 
      </article>
      <section
        aria-label="rating"
        class="_starsContainer_756783"
      >
        <svg
          aria-label="filled star"
          class="_starFilled_756783"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
        <svg
          aria-label="filled star"
          class="_starFilled_756783"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
        <svg
          aria-label="filled star"
          class="_starFilled_756783"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
        <svg
          aria-label="filled star"
          class="_starFilled_756783"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
        <svg
          aria-label="empty star"
          class="_starNotFilled_756783"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
          />
        </svg>
      </section>
      <p
        class="_publicInfoText_756783"
      >
        This journal entry is private
      </p>
    </section>
  </main>
</DocumentFragment>
`;
